### PR TITLE
[#161544435] Upgrade to cf-deployment v6.0.0

### PR DESCRIPTION
**Pull request by keymon**
What
----

We want to catchup our deployment and:

 - upgrade to cf-deployment v6.0.0 [1] (now we are on v4.5.0
 - upgrade the stemcell from Trusty to Xenial

We did not identify any major breaking change in the deployment.

During this upgrade:

 - We had to update the datadog agent to work on Xenial. See https://github.com/alphagov/paas-datadog-agent-boshrelease/pull/14
 - We had to create our own capi-release tarball (without changes) to workaround some issue with the upstream release. See https://github.com/cloudfoundry/cf-deployment/issues/663 and https://github.com/alphagov/paas-release-ci/pull/81

See commits for more info

[1] https://github.com/cloudfoundry/cf-deployment/releases/tag/v6.0.0

Related
--------

Bosh and stemcells upgrade: https://github.com/alphagov/paas-bootstrap/pull/210
and https://github.com/alphagov/paas-bootstrap/pull/214

How to review
-------------

This has been deployed on `towers` and `hector` dev environments.

We tested:
 - [x] deploy from scratch
 - [x] upgrade from master
 - [x] rotate certs (ca and leaf)
 - [x] rotate secrets

We run with an upgraded version of bosh.

Note: tehre were some errors of api availability, warnings due the stats
not being available. between 10 and 22. We assume them.

To review: Just double check the code and that the right stuff has been
deployed.

Who can review
--------------

Not me or @towers

### CODE REVIEW

File: manifests/cf-manifest/scripts/generate-manifest.sh [hunk 1]
```diff
@@ -56,7 +56,6 @@ bosh interpolate \
   --ops-file="${CF_DEPLOYMENT_DIR}/operations/enable-service-discovery.yml" \
   --ops-file="${PROMETHEUS_DEPLOYMENT_DIR}/manifests/operators/cf/add-prometheus-uaa-clients.yml" \
   ${slim_dev_deployment} \
-  --ops-file="${CF_DEPLOYMENT_DIR}/operations/experimental/disable-consul.yml" \
```
Comment (alext) [inline manifests/cf-manifest/scripts/generate-manifest.sh]: Now this is gone we can recombine `slim_dev_deployment` into `opsfile_args` (see https://github.com/alphagov/paas-cf/commit/e24a3177#diff-31c9fe7d5b430915c2e9db0a24e645c8)

File: concourse/pipelines/create-cloudfoundry.yml [hunk 2]
```diff
@@ -2128,24 +2118,7 @@ jobs:
...
-                      app['env'].merge!(env)
-                      app['routes'] = [
-                        { 'route' => 'metrics.${SYSTEM_DNS_ZONE_NAME}' }
-                      ]
-                    }
-                    File.write('manifest.yml', manifest.to_yaml)
-                  "
-
-                  cf zero-downtime-push paas-log-cache-adapter -f manifest.yml
+                  cf delete paas-log-cache-adapter -f
```
[the remark this answers is not in the document]
Comment (keymon) [reply concourse/pipelines/create-cloudfoundry.yml]: :+1: 

TBH, I am starting to think that I can fix the log-cache-adapter as part of this story. they only change the path of the endpoints, from `/v1/...` to `/api/v1/...`

File: manifests/cf-manifest/scripts/generate-manifest.sh (hunk 1 above)
Comment (keymon) [reply manifests/cf-manifest/scripts/generate-manifest.sh]: checking :thinking:

File: concourse/pipelines/create-cloudfoundry.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alext) [reply concourse/pipelines/create-cloudfoundry.yml]: Don't worry about that for now. I think the direction of travel is moving away from that it at the moment, so it's not worth investing effort in at the moment.

**Pull request merged**